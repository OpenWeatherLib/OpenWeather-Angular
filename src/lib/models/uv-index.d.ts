export interface UvIndex {
  lat: number;
  lon: number;
  date_iso: Date;
  date: Date;
  value: number;
}
