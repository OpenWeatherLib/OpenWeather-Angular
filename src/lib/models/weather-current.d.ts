import { Clouds } from "@lib/models/clouds";
import { Coordinates } from "@lib/models/coordinates";
import { Main } from "@lib/models/main";
import { Sys } from "@lib/models/sys";
import { WeatherPart } from "@lib/models/weather-part";
import { Wind } from "@lib/models/wind";

export interface WeatherCurrent {
    coord: Coordinates;
    weather: WeatherPart[];
    base: string;
    main: Main;
    visibility: number;
    wind: Wind;
    clouds: Clouds;
    dt: Date;
    sys: Sys;
    id: number;
    name: string;
    cod: number;
}
