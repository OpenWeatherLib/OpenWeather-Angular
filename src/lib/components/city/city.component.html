<mat-card class="ga-card" *ngIf="city">
  <mat-card-header>
    <img mat-card-avatar class="ga-card-header-image" src="{{cityPictureUrl}}" alt="City pic">
    <mat-card-title>City</mat-card-title>
    <mat-card-subtitle>{{city?.name}}</mat-card-subtitle>
  </mat-card-header>
  <img mat-card-image src="{{cityPictureUrl}}" alt="City pic">
  <mat-card-content>
    <p>Country: {{city?.country}}</p>
    <p>Population: {{city?.population}}</p>
    <p>Coordinates: Lat: {{city?.coord?.lat}}, Lon: {{city?.coord?.lon}}</p>
  </mat-card-content>
</mat-card>
