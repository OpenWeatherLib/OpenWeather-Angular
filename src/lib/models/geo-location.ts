import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class GeoLocation {
  lat: number = 720.0;
  long: number = 720.0;

  isDefault(): boolean {
    return this.lat === 720.0 && this.long === 720.0;
  }
}
