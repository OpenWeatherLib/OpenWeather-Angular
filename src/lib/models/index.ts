export * from "./address-component";
export * from "./city";
export * from "./city2";
export * from "./clouds";
export * from "./coordinates";
export * from "./coordinates2";
export * from "./geometry";
export * from "./main";
export * from "./rain";
export * from "./sys";
export * from "./weather-current";
export * from "./weather-forecast";
export * from "./weather-forecast-part";
export * from "./weather-part";
export * from "./wind";
export * from "./uv-index";
