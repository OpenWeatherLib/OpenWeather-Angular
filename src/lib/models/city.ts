import { GeoLocation } from "./geo-location";
import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class City {
  id: number = 0;
  name: string = "";
  country: string = "";
  population: number = 0;
  geoLocation: GeoLocation = new GeoLocation();

  isDefault(): boolean {
    return this.id === 0 && this.name === "" && this.country === "" && this.population === 0 && this.geoLocation.isDefault();
  }
}
