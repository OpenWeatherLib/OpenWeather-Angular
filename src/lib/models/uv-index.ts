import { GeoLocation } from "./geo-location";

export class UvIndex {
  geoLocation: GeoLocation = new GeoLocation();
  date: Date = new Date();
  value: number = 0.0;

  isDefault(): boolean {
    return this.geoLocation.isDefault() && this.value === 0.0;
  }
}
