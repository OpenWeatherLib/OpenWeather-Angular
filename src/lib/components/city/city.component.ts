import { Component, Input, OnInit } from "@angular/core";

import { BaseComponent } from "@lib/components/base-component/base.component";
import { City } from "@lib/models";
import { ImageService, OpenWeatherService } from "@lib/services";

@Component({
  selector: "ga-city",
  templateUrl: "./city.component.html",
  styleUrls: ["./city.component.scss"]
})
export class CityComponent extends BaseComponent implements OnInit {

  @Input() initialCityName: string = String().empty;

  city: City = null;
  cityPictureUrl: string = String().empty;

  constructor(private readonly imageService: ImageService,
    private readonly openWeatherService: OpenWeatherService) {
    super();
  }

  ngOnInit() {
    this.openWeatherService.loadCityData(this.initialCityName);

    this.registerSubscription(
      this.openWeatherService.city()
        .subscribe(city => {
          if (city) {
            this.city = city;
            this.imageService.receiveImagePictureUrl(this.city.name);
          }
        }));

    this.registerSubscription(
      this.imageService.cityPictureUrl()
        .subscribe(url => {
          this.cityPictureUrl = url;
        }));
  }
}
