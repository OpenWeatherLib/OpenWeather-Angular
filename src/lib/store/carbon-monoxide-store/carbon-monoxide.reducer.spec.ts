import { City, CarbonMonoxide } from "@lib/models";
import { loadCitySuccessAction } from "../city-store/city.actions";
import { loadCarbonMonoxideRequestAction, loadCarbonMonoxideSuccessAction, loadCarbonMonoxideErrorAction, setAccuracy, setDateTime } from "./carbon-monoxide.actions";
import { carbonMonoxideReducer } from "./carbon-monoxide.reducer";
import { CarbonMonoxideState } from "./carbon-monoxide.state";

describe("CarbonMonoxide Reducer Tests", () => {

    const initialState: CarbonMonoxideState = {
        carbonMonoxide: undefined,
        dateTime: "2019-07-12",
        accuracy: 2,
        isLoading: false,
        error: "Test Error"
    };

    describe("State Changes", () => {
        test("should have initial state after invalid action", () => {
            // Arrange & Act
            const testState: CarbonMonoxideState = carbonMonoxideReducer(initialState, { type: "INVALID_ACTION" } as any);

            // Assert
            expect(testState).toBe(initialState);
        });

        test("should load city on success of loading", () => {
            // Arrange
            const loadRequest = loadCarbonMonoxideRequestAction({ city: {} as City });
            const loadSuccess = loadCarbonMonoxideSuccessAction({ carbonMonoxide: {} as CarbonMonoxide });

            // Act
            const testState: CarbonMonoxideState = [loadRequest, loadSuccess].reduce(carbonMonoxideReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should save error on error of loading", () => {
            // Arrange
            const loadRequest = loadCarbonMonoxideRequestAction({ city: {} as City });
            const loadError = loadCarbonMonoxideErrorAction({ error: "Error" });

            // Act
            const testState: CarbonMonoxideState = [loadRequest, loadError].reduce(carbonMonoxideReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set loading to true on loadCitySuccessAction", () => {
            // Arrange
            const loadSuccess = loadCitySuccessAction({ city: {} as City });

            // Act
            const testState: CarbonMonoxideState = [loadSuccess].reduce(carbonMonoxideReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set accuracy on setAccuracy", () => {
            // Arrange
            const setAccuracyAction = setAccuracy({ accuracy: 7 });

            // Act
            const testState: CarbonMonoxideState = [setAccuracyAction].reduce(carbonMonoxideReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set dateTime on setDateTime", () => {
            // Arrange
            const setDateTimeAction = setDateTime({ dateTime: "DateTime" });

            // Act
            const testState: CarbonMonoxideState = [setDateTimeAction].reduce(carbonMonoxideReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });
    });
});
