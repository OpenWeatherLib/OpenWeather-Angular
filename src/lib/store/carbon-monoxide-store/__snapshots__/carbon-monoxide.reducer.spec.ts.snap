// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarbonMonoxide Reducer Tests State Changes should load city on success of loading 1`] = `
Object {
  "accuracy": 2,
  "carbonMonoxide": Object {},
  "dateTime": "2019-07-12",
  "error": undefined,
  "isLoading": false,
}
`;

exports[`CarbonMonoxide Reducer Tests State Changes should save error on error of loading 1`] = `
Object {
  "accuracy": 2,
  "carbonMonoxide": undefined,
  "dateTime": "2019-07-12",
  "error": "Error",
  "isLoading": false,
}
`;

exports[`CarbonMonoxide Reducer Tests State Changes should set accuracy on setAccuracy 1`] = `
Object {
  "accuracy": 7,
  "carbonMonoxide": undefined,
  "dateTime": "2019-07-12",
  "error": "Test Error",
  "isLoading": false,
}
`;

exports[`CarbonMonoxide Reducer Tests State Changes should set dateTime on setDateTime 1`] = `
Object {
  "accuracy": 2,
  "carbonMonoxide": undefined,
  "dateTime": "DateTime",
  "error": "Test Error",
  "isLoading": false,
}
`;

exports[`CarbonMonoxide Reducer Tests State Changes should set loading to true on loadCitySuccessAction 1`] = `
Object {
  "accuracy": 2,
  "carbonMonoxide": undefined,
  "dateTime": "2019-07-12",
  "error": "Test Error",
  "isLoading": true,
}
`;
